os-autoinst-openvswitch

```diff
@@ -33,10 +33,17 @@ sub init_switch {
     }
     system('ovs-vsctl', 'br-exists', $self->{BRIDGE});
 
-    my $bridge_conf = `ip addr show $self->{BRIDGE}`;
-
-    $self->{MAC} = $1 if $bridge_conf =~ /ether\s+(([0-9a-f]{2}:){5}[0-9a-f]{2})\s/;
-    $self->{IP}  = $1 if $bridge_conf =~ /inet\s+(([0-9]+.){3}[0-9]+\/[0-9]+)\s/;
+    my $timeout = 60;
+    do {
+        my $bridge_conf = `ip addr show $self->{BRIDGE}`;
+
+        $self->{MAC} = $1 if $bridge_conf =~ /ether\s+(([0-9a-f]{2}:){5}[0-9a-f]{2})\s/;
+        $self->{IP}  = $1 if $bridge_conf =~ /inet\s+(([0-9]+.){3}[0-9]+\/[0-9]+)\s/;
+        last if $self->{IP};
```

Comment:
I am not sure how to break this loop. I wrote a familiar Perl script file using `do {  last } until ()`, and when I execute it, report `Can't "last" outside a loop block at ... line`. @perlpunk could you have a look at it? I am not very familiar with `do { } until `. Thanks.

Reply:
you are right. Can't say how I missed this as I was sure I tested this code :D

Reply:
I actually completely avoid `do ... until` in perl.
But I have to admit that I don't understand why the `last` gives a warning, but it still seems to do the trick.

Doubling the `{}` will remove the warning:
```
do {{
  ...
  last if ...;
}} until ...;
```
But i'd still suggest we rather use a while loop here.

Reply:
Well, used an explicit for-loop which I hope you are also ok with :)